config-provisioning/src/main/java/com/yahoo/config/provision/NodeResources.java

```diff
@@ -145,33 +160,20 @@ private boolean lessThan(GpuResources other) {
 
         public GpuResources plus(GpuResources other) {
             if (other.isZero()) return this;
-            var thisMem = this.count() * this.memoryGiB();
-            var otherMem = other.count() * other.memoryGiB();
-            return new NodeResources.GpuResources(1, thisMem + otherMem);
+            if (this.isZero()) return other;
+            if (this.type != other.type) throw new IllegalArgumentException("Bad add: " + this + " + " + other);
+            double mem = Math.max(this.memoryGiB, other.memoryGiB);
```

Comment:
Are you changing the definition this way on purpose? The previous definition preserved the total memory of the two resources added, and since memory is what we charge for that does make sense. With this, 1 gpu with 8Gb memory + 10 with 4Gb become 11 with 8Gb memory. Wouldn't the minimal change you need here just be to only add the type verification?